Investigate, fix and re-enable feature-flags.spec.ts

<!--------- For bugs and general issues --------->
**Setup**
- Rancher version: 2.11.0
- Rancher UI Extensions: N/A
- Browser type & version: N/A

**Describe the bug**
<!--A clear and concise description of what the bug is.-->
We need to investigate the failures and re-enable feature-flags.spec.ts

Looks like there's a rancher crash when the user re-enables the harvester FF


```
E0304 11:51:11.318233      68 panic.go:262] "Observed a panic" panic="runtime error: invalid memory address or nil pointer dereference" panicGoValue="\"invalid memory address or nil pointer dereference\"" stacktrace=<
	goroutine 2569 [running]:
	k8s.io/apimachinery/pkg/util/runtime.logPanic({0xb5912f0, 0x10df3ae0}, {0x8a09d60, 0x10bcacf0})
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/runtime/runtime.go:107 +0xbc
	k8s.io/apimachinery/pkg/util/runtime.handleCrash({0xb5912f0, 0x10df3ae0}, {0x8a09d60, 0x10bcacf0}, {0x10df3ae0, 0x0, 0x10000c00e096a60?})
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/runtime/runtime.go:82 +0x5e
	k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00e096a60?})
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/runtime/runtime.go:59 +0x108
	panic({0x8a09d60?, 0x10bcacf0?})
		/usr/lib64/go/1.23/src/runtime/panic.go:785 +0x132
	github.com/rancher/rancher/pkg/controllers/management/feature.(*handler).syncHarvesterBaremetalContainerWorkloadFeature(0xc00b932780, 0xc00385c500?)
		/app/pkg/controllers/management/feature/feature_handler.go:155 +0xb1
	github.com/rancher/rancher/pkg/controllers/management/feature.(*handler).sync(0xc00b932780, {0xa4d4b40?, 0xc0063fcb01?}, 0x479f2d?)
		/app/pkg/controllers/management/feature/feature_handler.go:61 +0x105
	github.com/rancher/wrangler/v3/pkg/generic.(*Controller[...].func1({0xb522188?, 0xc00385c500?})
		/go/pkg/mod/github.com/rancher/wrangler/v3@v3.2.0-rc.3/pkg/generic/controller.go:169 +0x44
	github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xa4d4b40?, {0xc002c136c0?, 0xa63222c?}, {0xb522188?, 0xc00385c500?})
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/sharedcontroller.go:29 +0x32
	github.com/rancher/lasso/pkg/controller.(*SharedHandler).OnChange(0xc001258190, {0xc002c136c0, 0x9}, {0xb522188, 0xc00385c500})
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/sharedhandler.go:75 +0x202
	github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc0010a0370, {0xc002c136c0, 0x9})
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:236 +0x12e
	github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc0010a0370, {0x81027e0, 0xc00e096a60})
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:217 +0xd6
	github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc0010a0370)
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:194 +0x45
	github.com/rancher/lasso/pkg/controller.(*controller).runWorker(...)
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:183
	k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:226 +0x33
	k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00145d1e0, {0xb508de0, 0xc00230be00}, 0x1, 0xc001984540)
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:227 +0xaf
	k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00145d1e0, 0x3b9aca00, 0x0, 0x1, 0xc001984540)
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:204 +0x7f
	k8s.io/apimachinery/pkg/util/wait.Until(...)
		/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:161
	created by github.com/rancher/lasso/pkg/controller.(*controller).run in goroutine 2151
		/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:151 +0x354
 >
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x754d771]

goroutine 2569 [running]:
k8s.io/apimachinery/pkg/util/runtime.handleCrash({0xb5912f0, 0x10df3ae0}, {0x8a09d60, 0x10bcacf0}, {0x10df3ae0, 0x0, 0x10000c00e096a60?})
	/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/runtime/runtime.go:89 +0xee
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00e096a60?})
	/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/runtime/runtime.go:59 +0x108
panic({0x8a09d60?, 0x10bcacf0?})
	/usr/lib64/go/1.23/src/runtime/panic.go:785 +0x132
github.com/rancher/rancher/pkg/controllers/management/feature.(*handler).syncHarvesterBaremetalContainerWorkloadFeature(0xc00b932780, 0xc00385c500?)
	/app/pkg/controllers/management/feature/feature_handler.go:155 +0xb1
github.com/rancher/rancher/pkg/controllers/management/feature.(*handler).sync(0xc00b932780, {0xa4d4b40?, 0xc0063fcb01?}, 0x479f2d?)
	/app/pkg/controllers/management/feature/feature_handler.go:61 +0x105
github.com/rancher/wrangler/v3/pkg/generic.(*Controller[...].func1({0xb522188?, 0xc00385c500?})
	/go/pkg/mod/github.com/rancher/wrangler/v3@v3.2.0-rc.3/pkg/generic/controller.go:169 +0x44
github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xa4d4b40?, {0xc002c136c0?, 0xa63222c?}, {0xb522188?, 0xc00385c500?})
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/sharedcontroller.go:29 +0x32
github.com/rancher/lasso/pkg/controller.(*SharedHandler).OnChange(0xc001258190, {0xc002c136c0, 0x9}, {0xb522188, 0xc00385c500})
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/sharedhandler.go:75 +0x202
github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc0010a0370, {0xc002c136c0, 0x9})
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:236 +0x12e
github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc0010a0370, {0x81027e0, 0xc00e096a60})
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:217 +0xd6
github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc0010a0370)
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:194 +0x45
github.com/rancher/lasso/pkg/controller.(*controller).runWorker(...)
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:183
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:226 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00145d1e0, {0xb508de0, 0xc00230be00}, 0x1, 0xc001984540)
	/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:227 +0xaf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00145d1e0, 0x3b9aca00, 0x0, 0x1, 0xc001984540)
	/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:204 +0x7f
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.32.2/pkg/util/wait/backoff.go:161
created by github.com/rancher/lasso/pkg/controller.(*controller).run in goroutine 2151
	/go/pkg/mod/github.com/rancher/lasso@v0.2.1/pkg/controller/controller.go:151 +0x354
INFO: Running k3s server --cluster-init --cluster-reset
2025/03/04 11:51:26 [INFO] Rancher version v2.11-a2ff78325d1768b0a52fa65a88c00a1f3cddace5-head (a2ff78325d1768b0a52fa65a88c00a1f3cddace5) is starting
```

--- a/cypress/e2e/tests/pages/global-settings/feature-flags.spec.ts
+++ b/cypress/e2e/tests/pages/global-settings/feature-flags.spec.ts
@@ -7,7 +7,7 @@ import ClusterDashboardPagePo from '@/cypress/e2e/po/pages/explorer/cluster-dash
 const featureFlagsPage = new FeatureFlagsPagePo();
 const burgerMenu = new BurgerMenuPo();
 
-describe.skip('Feature Flags', { testIsolation: 'off' }, () => {
+describe('Feature Flags', { testIsolation: 'off' }, () => {
   before(() => {
     cy.login();
     HomePagePo.goTo();

